tests/fv/fv_test.go

```diff
@@ -81,6 +87,237 @@ var _ = Describe("PolicyController", func() {
 		apiserver.Stop()
 	})
 
+	Context("nodes", func() {
+		It("should be removed in response to a k8s node delete", func() {
+			kn := &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: kNodeName,
+				},
+			}
+			_, err := k8sClient.CoreV1().Nodes().Create(kn)
+			Expect(err).NotTo(HaveOccurred())
+
+			cn := &api.Node{
+				Metadata: api.NodeMetadata{
+					Name: cNodeName,
+				},
+				Spec: api.NodeSpec{
+					OrchRefs: []api.OrchRef{
+						{
+							NodeName: kNodeName,
+							Orchestrator: "k8s",
+						},
+					},
+				},
+			}
+			_, err = calicoClient.Nodes().Create(cn)
+			Expect(err).NotTo(HaveOccurred())
+
+			k8sClient.CoreV1().Nodes().Delete(kNodeName, &metav1.DeleteOptions{})
+			Eventually(func() *api.Node {
+				node, _ := calicoClient.Nodes().Get(api.NodeMetadata{Name: cNodeName})
+				return node
+			}, time.Second*2, 500*time.Millisecond).Should(BeNil())
+		})
+
+		It("should be removed if they reference a k8sNode that doesn't exist", func() {
+			cn := &api.Node{
+				Metadata: api.NodeMetadata{
+					Name: cNodeName,
+				},
+				Spec: api.NodeSpec{
+					OrchRefs: []api.OrchRef{
+						{
+							NodeName: "k8sfakenode",
+							Orchestrator: "k8s",
+						},
+					},
+				},
+			}
+			_, err := calicoClient.Nodes().Create(cn)
+			Expect(err).NotTo(HaveOccurred())
+
+			k8sClient.CoreV1().Nodes().Delete(kNodeName, &metav1.DeleteOptions{})
+			Eventually(func() *api.Node {
+				node, _ := calicoClient.Nodes().Get(api.NodeMetadata{Name: cNodeName})
+				return node
+			}, time.Second*2, 500*time.Millisecond).Should(BeNil())
```

Comment:
This one should stay at 15s.

Any test that relies on the reconciler needs a longer timeout, any test that doesn't needs a short one.